spec/models/validations/sales/financial_validations_spec.rb

```diff
@@ -339,4 +339,244 @@
+            expect(record.errors["grant"]).to include("Mortgage, deposit, and grant total must equal £18,000.00")
+          end
+
+          it "returns false if mortgage, grant and deposit total equals market value - discount" do
+            record.mortgage = 10_000
+            financial_validator.validate_discounted_ownership_value(record)
+            expect(record.errors["mortgageused"]).to be_empty
+            expect(record.errors["mortgage"]).to be_empty
+            expect(record.errors["value"]).to be_empty
+            expect(record.errors["deposit"]).to be_empty
+            expect(record.errors["ownershipsch"]).to be_empty
+            expect(record.errors["discount"]).to be_empty
+            expect(record.errors["grant"]).to be_empty
+          end
+        end
+      end
+
+      context "when mortgage is not routed to" do
+        let(:record) { FactoryBot.build(:sales_log, mortgageused: 2, deposit: 5_000, grant: 3_000, value: 20_000, discount: 10, ownershipsch: 2, saledate: now) }
+
+        it "returns true if grant and deposit total does not equal market value - discount" do
+          financial_validator.validate_discounted_ownership_value(record)
+          expect(record.errors["mortgageused"]).to include("Mortgage, deposit, and grant total must equal £18,000.00")
+          expect(record.errors["mortgage"]).to include("Mortgage, deposit, and grant total must equal £18,000.00")
+          expect(record.errors["value"]).to include("Mortgage, deposit, and grant total must equal £18,000.00")
+          expect(record.errors["deposit"]).to include("Mortgage, deposit, and grant total must equal £18,000.00")
+          expect(record.errors["ownershipsch"]).to include("Mortgage, deposit, and grant total must equal £18,000.00")
+          expect(record.errors["discount"]).to include("Mortgage, deposit, and grant total must equal £18,000.00")
+          expect(record.errors["grant"]).to include("Mortgage, deposit, and grant total must equal £18,000.00")
+        end
+
+        it "returns false if mortgage, grant and deposit total equals market value - discount" do
+          record.grant = 13_000
+          financial_validator.validate_discounted_ownership_value(record)
+          expect(record.errors["mortgageused"]).to be_empty
+          expect(record.errors["mortgage"]).to be_empty
+          expect(record.errors["value"]).to be_empty
+          expect(record.errors["deposit"]).to be_empty
+          expect(record.errors["ownershipsch"]).to be_empty
+          expect(record.errors["discount"]).to be_empty
+          expect(record.errors["grant"]).to be_empty
+        end
+      end
+
+      context "when ownership is not discounted" do
+        let(:record) { FactoryBot.build(:sales_log, mortgage: 10_000, deposit: 5_000, grant: 3_000, value: 20_000, discount: 10, ownershipsch: 1, saledate: Time.zone.local(2023, 4, 1)) }
```

Comment:
This one should be in 2024, shouldn't it?

Reply:
Ah, yes, good spot!